Patch CACTUS AMC13 Makefiles to work on Debian/Ubuntu

lmoureaux opened the pull request:
The --as-needed linker flag (enabled by default on Ubuntu 18.04) breaks
compilation. Fix it by reordering -l flags.

Tested by doing the following:

```sh
$ rm -r thirdparty/cactus/amc13/
$ XDAQ_PLATFORM=x86_64 XDAQ_ROOT=$(readlink -f install/xdaq/x86_64/) make
```

Compilation succeeds.

lpetre-ulb commented on cmake/External_cactus.cmake:
```diff
@@ -24,12 +24,14 @@ ExternalProject_Add(cactus-amc13
   BUILD_COMMAND make
     "CACTUS_ROOT=${CMAKE_BINARY_DIR}/install/cactus"
     "INSTALL_PREFIX=<INSTALL_DIR>"
+    "SHELL=bash"
```
Have you tried you your patch in a clean build ? I'm a bit surprised that you don't need to pass `"SHELL=bash"` to the `INSTALL_COMMAND`. Or maybe that the directories are already created by `cactus-ipbus` ?

lmoureaux commented:
Rebased, added asyncresolv patch, tested clean build.

Make commands used:
```sh
XDAQ_PLATFORM=x86_64 make # I'm afraid of setting XDAQ_ROOT when building xDAQ
XDAQ_PLATFORM=x86_64 XDAQ_ROOT=$(readlink -f install/xdaq/x86_64/) make
```

lpetre-ulb commented:
Ok, thank you for addressing my concerns.

> ``` shell
> XDAQ_PLATFORM=x86_64 make # I'm afraid of setting XDAQ_ROOT when building xDAQ
> XDAQ_PLATFORM=x86_64 XDAQ_ROOT=$(readlink -f install/xdaq/x86_64/) make
> ```

You are right, setting `XDAQ_ROOT` during the build of `xDAQ` might failed the compilation.

lpetre-ulb commented:
Fixes #3.
